src/main/java/org/cryptomator/cryptofs/health/shortened/ShortenedNamesCheck.java

```diff
@@ -103,6 +115,38 @@ void checkShortenedName(Path dir) throws IOException {
 			}
 		}
 
+
+		/**
+		 * Determines if the string stored inside the name file is a base64url encoded ending with {@value Constants#CRYPTOMATOR_FILE_SUFFIX}.
+		 *
+		 * <em>visible for testing</em>
+		 *
+		 * @return {@link SyntaxResult} indicating if it is valid, invalid or affected by https://github.com/cryptomator/cryptofs/issues/121
+		 */
+		SyntaxResult checkSyntax(String toAnalyse) {
+			int posObligatoryC9rString = toAnalyse.indexOf(Constants.CRYPTOMATOR_FILE_SUFFIX);
+			if(posObligatoryC9rString == -1) {
+				return SyntaxResult.INVALID;
+			}
+
+			var encryptedFileName = toAnalyse.substring(0, posObligatoryC9rString);
+			if(!BASE64URL.canDecode(encryptedFileName)) {
+				return SyntaxResult.INVALID;
+			}
+
+			if (toAnalyse.substring(posObligatoryC9rString).length() > Constants.CRYPTOMATOR_FILE_SUFFIX.length() ) {
+				return SyntaxResult.TRAILING_BYTES;
+			}
```

Comment:
just a minor optimization to avoid creating substrings

```suggestion
			if (toAnalyse.length() > posObligatoryC9rString +  Constants.CRYPTOMATOR_FILE_SUFFIX.length()) {
				return SyntaxResult.TRAILING_BYTES;
			}

			var encryptedFileName = toAnalyse.substring(0, posObligatoryC9rString);
			if(!BASE64URL.canDecode(encryptedFileName)) {
				return SyntaxResult.INVALID;
			}
```

Reply:
The optimization is actually a change, which changes the outcome: Before, you always check the if the first part is base64url encoded and if it is not always return `INVALID` and only if it is valid, it might be `TRAILING_BYTES` result. 
If applying the change, we always check if there are trailing bytes. And only if there are none, it might return `INVALID`.

I decided against this change, since i see no benefit in fixing trailing bytes if the result would still be broken from _this checks perspective_.

Reply:
even without reordering, note the unnecessary substring, though:

```diff
-			if (toAnalyse.substring(posObligatoryC9rString).length() > Constants.CRYPTOMATOR_FILE_SUFFIX.length() ) {
+			if (toAnalyse.length() > posObligatoryC9rString +  Constants.CRYPTOMATOR_FILE_SUFFIX.length()) {
```

Reply:
the next time 🙂

Reply:
done in fe23ff483c26484c4e6a456b7d6797596f689b02